flow360/component/flow360_params/params_base.py

```diff
@@ -385,18 +393,12 @@ def _swap_key_in_nested_dict(cls, dictionary, key_to_replace, replacement_key):
         return dictionary
 
     @classmethod
-    def _generate_schema_for_optional_objects(cls, model: dict, key: str):
-        field = model["properties"].pop(key)
+    def _generate_schema_for_optional_objects(cls, schema: dict, key: str):
+        field = schema["properties"].pop(key)
         if field is not None:
-            ref = field.get("$ref")
-            if ref is None:
-                raise Flow360RuntimeError(
-                    f"Trying to apply optional field transform to a non-ref field {key}"
-                )
-
-            toggle_name = f"_add{key[0].upper() + key[1:]}"
+            toggle_name = _optional_toggle_name(key)
 
-            model["properties"][toggle_name] = {
+            schema["properties"][toggle_name] = {
                 "title": toggle_name[1:],
```

Comment:
we can now remove "add <field name>" convention here as the new UI design will have "add" button. The title can remain just the original property key